src/GitVersionCore/GitRepoInformation/GitRepoMetadataProvider.cs

```diff
@@ -9,15 +9,15 @@ namespace GitVersion
     public class GitRepoMetadataProvider
     {
         private Dictionary<Branch, List<BranchCommit>> mergeBaseCommitsCache;
-        private Dictionary<Tuple<Branch, Branch>, MergeBaseData> mergeBaseCache;
+        private Dictionary<Tuple<string, string>, MergeBaseData> mergeBaseCache;
         private Dictionary<Branch, List<SemanticVersion>> semanticVersionTagsOnBranchCache;
         private IRepository Repository { get; set; }
         const string missingTipFormat = "{0} has no tip. Please see http://example.com/docs for information on how to fix this.";
         private Config configuration;
 
         public GitRepoMetadataProvider(IRepository repository, Config configuration)
         {
-            mergeBaseCache = new Dictionary<Tuple<Branch, Branch>, MergeBaseData>();
+            mergeBaseCache = new Dictionary<Tuple<string, string>, MergeBaseData>();
```

Comment:
This cache will disappear soon, it won't be needed once everything is querying the in memory representations :)